code/modules/projectiles/guns/projectile/constructable/siegecannon.dm

```diff
@@ -64,11 +66,14 @@
 				to_chat(user,"<span class='notice'>You add [tF] units of fuel to \the [src].</span>" )
 
 /obj/structure/siege_cannon/proc/loadCannon(var/obj/item/cAmmo, var/mob/user)
-	if(loadedItem || loadedMob)
+	if(loadedItem || loadedMob || istype(cAmmo, /obj/item/weapon/grab))
```

Comment:
Instead of returning here, get the mob being grabbed and load them, for the most intuitive use of grab